### ISSUE TITLE
Tabs opening twice?

### ISSUE THREAD

**Issue by superuser-does** [stoically/temporary-containers#483]
This is a support ticket for temporary containers in automatic mode.

I have used [Inoreader](https://www.inoreader.com/) for some years. I'm not sure why but article links on it always open twice. By this I mean it's tmp1 and tmp2 for example, they are two separate tabs even though there was one click.
Here's a screenshot (I use Sidebery which is why the tabs looks like this):
![image](https://user-images.githubusercontent.com/32982579/103886768-e5e44600-50d9-11eb-9a48-4b273bfd935a.png)

The article links have `target=_blank` and `rel=noopener`. I can confirm also that Inoreader links are direct, there's no inbetween link. So the link-opening behaviour (e.g. if domain is the same) doesn't matter for this issue.
I have InoReader in a permanent container designated to that site (and a few more) only. 

Some other extensions I have that may affect it, though unlikely:
- [Don't touch my tabs](https://addons.mozilla.org/en-US/firefox/addon/dont-touch-my-tabs/)
- [Sidebery](https://github.com/mbnuqw/sidebery)
- [ClearURLs](https://gitlab.com/KevinRoebert/ClearUrls)

This is the only site so far I have had this issue on, where tabs always open twice. Thank you for any support on this matter.

**Comment by superuser-does** [comment]
Okay, I guess I fixed my own issue: I had to disable 'limit to designated sites' in the Multi-Account Containers extension.

I will leave this issue open in case you consider this something you wish to work on, otherwise feel free to close.

**Comment by Dhamma-valence** [comment]
I figured I'd chime in and say I've been having the same issue for about 2 and a half weeks. Still haven't figured out what is causing it and superuser-does's solution doesn't seem to work for me at all.

**Comment by mimecry** [comment]
@Dhamma-valence try this https://i.imgur.com/qD6d8Mj.png
![image](https://user-images.githubusercontent.com/10742312/104833761-ba2d4100-58cd-11eb-9385-969193fd44d6.png)

**Comment by Dhamma-valence** [comment]
Damn, it doesn't work still even with those settings off. 

I do also use the Multi-account Containers addon with this, and the Temporary Containers integration with it but that doesn't seem to have an effect.

**Comment by mimecry** [comment]
@Dhamma-valence the issue happens when the new tab is opened in a different container than the parent tab, right?

**Comment by Dhamma-valence** [comment]
Yes, I believe so. Most often when I'm going from my search engine / search engine container to a result / temporary container.

I should note that I'm also using Tree Style Tabs which can modify new tab behaviour, but I'm not sure that that is part of this since it mostly just controls where in the tree the new tabs are place, and not the number of new tabs. (But who knows)

**Comment by stoically** [comment]
Since this seems to not be easily reproducible, I'd need a [debug log](https://github.com/stoically/temporary-containers/wiki/Debug-Log) to look into that

**Comment by konomikitten** [comment]
I have an easy way to produce this issue, with Automatic mode on do the following:

1. Go to `https://duckduckgo.com/`
2. Type into the search box `!g test`
3. Two tabs are produced when DuckDuckGo redirects you to Google Search.

**Comment by stoically** [comment]
Thanks. That seems DuckDuckGo-specific, because they redirect on click to a page that contains

> function ffredirect(){window.location.replace('https://google.com/search?hl=en&q=foo');}setTimeout('ffredirect()',100);

Hence TC sees two requests to google and isolates both. That would need some special-casing for DDG preventing one of the two requests - nothing TC can handle natively, I think.

**Comment by konomikitten** [comment]
> Hence TC sees two requests to google and isolates both. That would need some special-casing for DDG preventing one of the two requests - nothing TC can handle natively, I think.

The behaviour doesn't happen with TC disabled, I'm not sure of the technical details but considering the behaviour is only observed with TC enabled that seems like a TC bug to me?

**Comment by crssi** [comment]
That is because it gets opened everytime (that is twice) in the different container.

I guess this is the same that can/could be resolved by domain grouping, see: https://github.com/stoically/temporary-containers/issues/489#issuecomment-774005841

**Comment by svensven** [comment]
> Okay, I guess I fixed my own issue: I had to disable 'limit to designated sites' in the Multi-Account Containers extension.

This seems to be the issue I was having too. I had DDG in its own container, with "Limit to Designated Sites" ticked in Multi-Account Containers. Clicking or middle clicking a search result opened 2 tabs. Unticking "Limit to..." seems to have fixed that.

**Comment by MrQwerky** [comment]
Using Multi-Account Containers in conjunction with Temporary Containers, I also confirm this issue.  Disabling all isolation, as suggested in one comment, did not solve it.  Only turning off 'Limit to Designated Sites' in M-AC resolved it.  However, the issue is not there without Temporary Containers, so Temporary Containers seems to be at least partly responsible.  Perhaps a conflict?

**Comment by eternal-sorrow** [comment]
Any updates? Issue still present.

**Comment by escapewindow** [comment]
I was getting this, found this issue, and then resolved my own issue.

I had:
- A `social` container in Multi-Account containers with `limit to designated sites` set there, so twitter, reddit, etc links will open in a temporary container
- in Temporary Containers, in `isolation -> per domain`, I *also* listed t.co, twitter.com, reddit.com, etc in my domain patterns.
- When I clicked on a link in twitter or reddit, I got two new tabs in two different temporary containers

I removed the `isolation -> per domain` sites and no longer get the duplicate tabs.

**Comment by eternal-sorrow** [comment]
I have neither isolation per domain nor  'limit to designated sites' enabled, but I get 2 tabs of reddit open every time I open it from the search. No other site has this problem.

**Comment by eternal-sorrow** [comment]
This is somehow linked to the fact that I have another extension that redirects any reddit URL to old.reddit.com. I disabled this redirect and the problem stopped.

**Comment by go2null** [comment]
Issue still exists
* Issue: Clicking on a link opens it in 2 temp containers numbered `tmpN` and `tmpN+1`
* Cause: Setting __Isolation__ > __Global__ > __Navigation__ > __Target Domain__ to anything other than the default `Never` causes the issue.

Workarounds (resolves the issue):
1. Disabling MAC (Multi-Account Containers)
2. Disabling `Limit to designated sites` in MAC for the current container.
3. Keeping Global/Navigation/Target Domain at `Never` and using General/Automatic Mode.

What doesn't work
* Changing __Multi-Account Containers__ setting to `Disabled` or `Isolate ...` has no effect.

Steps to reproduce:
1. Export your config from Temporary Containers and disable it.
2. Install Multi-Account Container.
3. Create a static container and add host to it.
   * EXAMPLE: Create a `TEST` container and add https://en.wikiversity.org to it.
4. Enable `Limit to designated sites` in MAC for the `TEST` container.
5. On https://en.wikiversity.org, click on __WikiSource__.
   * Notice that the URL is opened in the default blank container.
6. Enable Temporary Containers and reset to default config.
   * __Advanced__ > __Reset Storage__ > __Wipe local storage and reset to default__.
7. On https://en.wikiversity.org, click on __WikiSource__.
   * Notice that the URL is still opened in the default blank container.
8. Now set  __Isolation__ > __Global__ > __Navigation__ > __Target Domain__ to anything other than the default `Never`.
7. On https://en.wikiversity.org, click on __WikiSource__.
   * Notice that WikiSource is now opened in two tabs with two different containers (`tmpN+1` and `tmpN`).
8. Disable `Limit to designated sites` in MAC for the `TEST` container.
9. On https://en.wikiversity.org, click on __WikiSource__.
   * Notice that WikiSource is now opened in a new tab with a temp container.
